samples/net/sockets/echo_async_select/sample.yaml

```diff
@@ -2,8 +2,15 @@ sample:
   description: BSD Sockets API TCP echo server sample
     using non-blocking sockets
   name: socket_echo_async
+common:
+  harness: net
+  platform_whitelist: qemu_x86
+  tags: net socket
 tests:
   sample.net.sockets.echo_async_select:
-    harness: net
-    platform_whitelist: qemu_x86
-    tags: net socket
+    extra_configs:
+      - CONFIG_NET_SOCKETS_POSIX_NAMES=y
```

Comment:
Interesting - so you can use `! IS_ENABLED(CONFIG_NET_SOCKETS_POSIX_NAMES)` and then call the Zephyr API directly. Is that just a compile test or are there some other checks performed?

Reply:
> you can use ! IS_ENABLED(CONFIG_NET_SOCKETS_POSIX_NAMES)

In principle, yes. Except that you likely shouldn't test CONFIG_NET_SOCKETS_POSIX_NAMES. Normally, you should use either CONFIG_POSIX_API=y or CONFIG_NET_SOCKETS_POSIX_NAMES=y.

Well, tests may be an exception, but here we're talking about extending tests to cover CONFIG_POSIX_API=y. Having 3 test configs is definitely too much, so I'd suggest to concentrate on CONFIG_POSIX_API=y (which conflicts with CONFIG_NET_SOCKETS_POSIX_NAMES=y, and thus requires it to be =n) and CONFIG_NET_SOCKETS_POSIX_NAMES=y. Both these variants provide names like `recv`. CONFIG_POSIX_API=y also provides name like `close` and `write`. (Well, as you discovered, write is also available w/o it, it comes directly from newlib then. Yeah, there're a lot to improve and cleanup.)

> Is that just a compile test or are there some other checks performed?

[sanitycheck](https://docs.zephyrproject.org/latest/guides/test/sanitycheck.html) is Zephyr's testing tool, it can both build apps and run them in QEMU/on real hardware. Of course, quite not all samples are actually run. Improving that is subject of work of various stakeholders.